packages/desktop/tracker.js

```diff
@@ -0,0 +1,69 @@
+import activeWin from 'active-win';
+import fs from 'fs-extra';
+import _ from 'lodash';
+import path from 'path';
+
+//const __dirname = path.resolve();
+//const filePath = path.join(__dirname, "activites.json")
+//const interval = 2000
+
+class ActiveWindowObject {
+	constructor(filePath, interval) {
+		(this.startTime = null), //Storing the start time of the active window
+			(this.app = null); //Collecting data of the window which will be active
+		this.filePath = filePath;
+		this.interval = interval;
+	}
+
+	async storeTime() {
+		const file = await fs.readJSON(this.filePath);
+		let endTime = new Date();
+		let startTime = this.startTime;
+
+		//writing the json file
+		const {
+			owner: { name },
+			title,
+			url,
+		} = this.app;
+		_.defaultsDeep(file, { [name]: { [title]: { timeSpent: 0, url } } });
```

Comment:
You can assign defaults without lodash using [??](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator)
```js
file[name] = file[name] ?? {};
file[name][title] = file[name][title] ?? { timeSpent: 0, url };
```